tests/lib/rules/hook-use-state-names.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @fileoverview Ensure symmetric naming of setState hook value and setter variables
+ * @author Duncan Beevers
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/hook-use-state-names');
+const parsers = require('../../helpers/parsers');
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module'
+  }
+});
+
+ruleTester.run('hook-set-state-names', rule, {
+  valid: [
+    {
+      code: 'const [color, setColor] = useState()'
+    },
+    {
+      code: 'const [color, setColor] = useState(\'#ffffff\')'
+    },
+    {
+      code: 'const [color, setColor] = React.useState()'
+    },
+    {
+      code: 'const [color, setColor, unused] = useState()'
+    },
+    {
+      code: 'const [color1, setColor1] = useState()'
+    },
+    {
+      code: 'const [, setColor] = useState()'
+    },
+    {
+      code: 'const useStateResult = useState()'
```

Comment:
what about if someone does `const [state, setThatState] = useStateResult;` or `const state = useStateResult[0]; const setThatState = useStateResult[1];` on the next lines?

Reply:
This rule only looks for calls to `useState` which are directly destructured+assigned;

These intermediate-value forms aren't recognized by this rule.

Reply:
This is now an error as well.